@vvssttkk, fixed few failed tests I was talking about. Please investigate others. E.g. one of the frequent failures I reproduced in 3.7:

```
tmp %f $ python                                                                                                                                   181ms
Python 3.7.10 (default, Jun 25 2021, 16:52:37)
[Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import hashlib
>>> hashlib.md5()
<md5 HASH object @ 0x109a1bdb0>
>>> hashlib.md5(userforsecurity=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: openssl_md5() takes no keyword arguments
>>>
```

--- a/mlem/core/artifacts.py
+++ b/mlem/core/artifacts.py
@@ -323,7 +323,7 @@ class LocalArtifact(FSSpecArtifact):
 
 
 def md5_fileobj(fobj):
-    hash_md5 = hashlib.md5(usedforsecurity=False)
+    hash_md5 = hashlib.md5()  # nosec: B324
     for chunk in iter(lambda: fobj.read(CHUNK_SIZE), b""):
         hash_md5.update(chunk)
     return hash_md5.hexdigest()

--- a/mlem/core/objects.py
+++ b/mlem/core/objects.py
@@ -302,8 +302,8 @@ class MlemObject(MlemABC):
             self._write_meta(self.location)
 
     def meta_hash(self):
-        return hashlib.md5(
-            safe_dump(self.dict()).encode("utf8"), usedforsecurity=False
+        return hashlib.md5(  # nosec: B324
+            safe_dump(self.dict()).encode("utf8")
         ).hexdigest()
 
 

--- a/mlem/utils/templates.py
+++ b/mlem/utils/templates.py
@@ -2,7 +2,12 @@ from typing import ClassVar, List
 
 from fsspec import AbstractFileSystem
 from fsspec.implementations.local import LocalFileSystem
-from jinja2 import Environment, FileSystemLoader, StrictUndefined, select_autoescape
+from jinja2 import (
+    Environment,
+    FileSystemLoader,
+    StrictUndefined,
+    select_autoescape,
+)
 from pydantic import BaseModel
 
 

--- a/mlem/utils/templates.py
+++ b/mlem/utils/templates.py
@@ -2,7 +2,7 @@ from typing import ClassVar, List
 
 from fsspec import AbstractFileSystem
 from fsspec.implementations.local import LocalFileSystem
-from jinja2 import Environment, FileSystemLoader, StrictUndefined
+from jinja2 import Environment, FileSystemLoader, StrictUndefined, select_autoescape
 from pydantic import BaseModel
 
 
@@ -22,7 +22,7 @@ class TemplateModel(BaseModel):
         j2 = Environment(
             loader=FileSystemLoader(self.templates_dir + [self.TEMPLATE_DIR]),
             undefined=StrictUndefined,
-            autoescape=True,
+            autoescape=select_autoescape(),
         )
         template = j2.get_template(self.TEMPLATE_FILE)
         args = self.prepare_dict()

